### ISSUE TITLE
Support flat array input format.

### ISSUE THREAD

**Issue by mbostock** [d3/d3-delaunay#27]
Now that mapbox/delaunator#17 is fixed, we should support the same API for the Delaunay constructor. Or, if we want to retain the current private copy constructor API, we’d need to expose a new factory method. Delaunay.of(*array*) comes to mind, but that seems like a bad idea—that should be reserved for Delaunay.of(*…values*) for consistency with Array.of. So some options:

1. Remove the copy constructor and just do new Delaunay(*array*). Pros: simple. Cons: copying a Delaunay instance would then require Object.create, but I don’t really anticipate this being a common pattern anyway; requires the `new` operator.

2. Rename Delaunay.from to Delaunay.fromPoints and introduce a new Delaunay.from that takes the flat array. Pros: retains copy constructor. Cons: confusing terminology, as both constructors accept “points”.

3. Add Delaunay.fromFlat(*array*), and keep existing constructor and Delaunay.from. Pros: does everything. Cons: fromFlat is the ugliest name for the use case we want to encourage.

4. Detect in Delaunay.from whether the given *array* is flat or not. Pros: “just works”. Cons: sometimes it doesn’t, and overloading by type is icky.

Feels like the first option is the “obvious” choice, but wanted to consider the others.

**Comment by mourner** [comment]
How about ditching the private copy constructor and simply exposing the original Delaunator class with an augmented prototype (adding methods like `voronoi`)?

**Comment by mbostock** [comment]
That’s what I did originally, but I enjoy the encapsulation provided by a separate class. The Delaunator instance exposes some public fields that aren’t necessary for my use cases (*delaunator*.ids, *delaunator*.trianglesLen). And a separate representation for the constructed triangulation allows freeing of temporary state used during construction (mainly *delaunator*._hash, but also *delaunator*._cx, *delaunator*._cy). And it encapsulates the private methods.

**Comment by mbostock** [comment]
I’ve implemented the first strategy in #25. I’ve also augmented Delaunay.from to more closely match the behavior of Array.from.

**Comment by mbostock** [comment]
Here’s a fun example:

```js
Delaunay.from({length: 100}, () => Math.random() * width, () => Math.random() * height)
```
